core/devmode/src/main/java/io/quarkus/dev/DevModeMain.java

```diff
@@ -70,8 +71,17 @@ public static void main(String... args) throws Exception {
                 System.setProperty(i.getKey(), i.getValue());
             }
         }
-        //the path that contains the compiled classes
-        classesRoot = new File(args[0]);
+
+        //a comma separated string containing the path to the compiled class and the path the resources output
+        if (args[0].contains(",")) {
+            final String[] parts = args[0].split(",");
+            classesRoot = new File(parts[0]);
+            resourcesOutputRoot = new File(parts[1]);
+        } else {
+            //the path that contains the compiled classes
+            classesRoot = new File(args[0]);
```

Comment:
Can't say I'm excited to pass 2 different values with the same parameter.

I think I would find it less of a problem if we made `classesRoot` a `List` called `classesRoots` and just manage it as the general case.

We would have only one root in the Maven case and 2 in the Gradle case. That would simplify the code, and moreover make it less hackyish.

WDYT?

Reply:
I like the idea, I think it would clean things up. I'll update the PR today or tomorrow

Reply:
Fixed